### PR TITLE
Document Affinity & CloudEP

### PR DESCRIPTION
**Pull request by RandySheriffH**
Document two features for coming 1.14 release:

1. Setting intra-op thread affinities.
2. CloudEP for AML hybrid inferencing.

### CODE REVIEW

File: docs/build/eps.md [hunk 1]
```diff
@@ -802,3 +802,32 @@ See more information on the CANN Execution Provider [here](../execution-provider
...
+## Cloud
+
+See the [Cloud Execution Provider](../execution-providers/Cloud-ExecutionProvider.md) documentation for more details.
+
+### Prerequisites
+
+For Linux, before bulding please:
+
+* install openssl dev package into the system, which is openssl-dev for redhat and libssl-dev for ubuntu.
+* if got multiple openssl dev versions installed in the system, please set environment variable "OPENSSL_ROOT_DIR" to the desired version, for example:
```
Comment (jeffmend) [inline docs/build/eps.md:815]: should be: "if you have multiple versions"

File: docs/execution-providers/Cloud-ExecutionProvider.md [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+title: Cloud
+description: Instructions to infer an ONNX model remotely with a cloud endpoint
+parent: Execution Providers
+nav_order: 12
+redirect_from: /docs/reference/execution-providers/Cloud-ExecutionProvider
+---
+
+# Cloud Execution Provider
+
+The Cloud Execution Provider enables ONNX Runtime to invoke a cloud endpoint for inferenece, the endpoint must be deployed beforehand.
```
Comment (pranavsharma) [inline docs/execution-providers/Cloud-ExecutionProvider.md:11]:
```suggestion
The Cloud Execution Provider enables ONNX Runtime to inference a model in the cloud. The model needs to be deployed to the cloud before using ORT.
```

Comment (pranavsharma) [reply docs/execution-providers/Cloud-ExecutionProvider.md:11]: Also, please explicitly mention the use case (small/big model) this preview is targeting.

File: docs/execution-providers/Cloud-ExecutionProvider.md [hunk 3]
```diff
@@ -0,0 +1,68 @@
+---
+title: Cloud
+description: Instructions to infer an ONNX model remotely with a cloud endpoint
+parent: Execution Providers
+nav_order: 12
+redirect_from: /docs/reference/execution-providers/Cloud-ExecutionProvider
+---
+
+# Cloud Execution Provider
```
Comment (pranavsharma) [inline docs/execution-providers/Cloud-ExecutionProvider.md:9]: Better to mention Preview in parentheses here.

File: docs/execution-providers/Cloud-ExecutionProvider.md [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+
+The Cloud Execution Provider enables ONNX Runtime to invoke a cloud endpoint for inferenece, the endpoint must be deployed beforehand.
+Cloud Execution Provider is in preview stage, all API(s) and usage are subjuct to change.
+
+## Contents
+
+To consume the endpoint, ONNX Runtime must load the same model locally, customer could then use configured run options to choose to go remote.
```
Comment (pranavsharma) [inline docs/execution-providers/Cloud-ExecutionProvider.md:16]: "same model" is misleading. We need to expand this by mentioning the use case being targeted in this preview.

File: docs/execution-providers/Cloud-ExecutionProvider.md [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+By far, Cloud Execution Provider is limited to:
+* only support [triton](https://github.com/triton-inference-server) server on [AML](https://learn.microsoft.com/en-us/azure/machine-learning/how-to-deploy-with-triton?tabs=python%2Cendpoint).
+* only build and run on Windows and Linux.
```
Comment (pranavsharma) [inline docs/execution-providers/Cloud-ExecutionProvider.md:19]: Also, only available as a Python pkg.
Also, this section should be titled ```Limitations (Preview)```

File: docs/performance/tune-performance.md [hunk 6]
```diff
@@ -169,7 +169,39 @@ Currently, there are no special provisions to employ mimalloc on Linux. It is re
 
 ### Thread management
 
-* Use the appropriate ORT API to set intra and inter op num threads. Inter op num threads is only used when parallel execution is enabled.
+#### Set number of intra-op threads
```
Comment (pranavsharma) [inline docs/performance/tune-performance.md:172]: We should add a separate section to call out support for NUMA.

**Comment by faxu** [comment]
@RandySheriffH please stage your changes for visual preview - https://github.com/microsoft/onnxruntime/wiki/Update-ONNX-Runtime-Docs#onnxruntimeaidocs

File: docs/execution-providers/Cloud-ExecutionProvider.md [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+
+# Cloud Execution Provider
+
+The Cloud Execution Provider enables ONNX Runtime to invoke a cloud endpoint for inferenece, the endpoint must be deployed beforehand.
+Cloud Execution Provider is in preview stage, all API(s) and usage are subjuct to change.
+
+## Contents
+
+To consume the endpoint, ONNX Runtime must load the same model locally, customer could then use configured run options to choose to go remote.
+By far, Cloud Execution Provider is limited to:
```
[the remark this answers is not in the document]
Comment (RandySheriffH) [reply docs/execution-providers/Cloud-ExecutionProvider.md]: Corrected, thanks!

File: docs/build/eps.md (hunk 1 above)
Comment (RandySheriffH) [reply docs/build/eps.md]: Done.

File: docs/execution-providers/Cloud-ExecutionProvider.md (hunk 2 above)
Comment (RandySheriffH) [reply docs/execution-providers/Cloud-ExecutionProvider.md]: Added small-big model case.

File: docs/execution-providers/Cloud-ExecutionProvider.md (hunk 3 above)
Comment (RandySheriffH) [reply docs/execution-providers/Cloud-ExecutionProvider.md]: Done.

File: docs/execution-providers/Cloud-ExecutionProvider.md (hunk 4 above)
Comment (RandySheriffH) [reply docs/execution-providers/Cloud-ExecutionProvider.md]: Corrected.

File: docs/execution-providers/Cloud-ExecutionProvider.md (hunk 5 above)
Comment (RandySheriffH) [reply docs/execution-providers/Cloud-ExecutionProvider.md]: Done.

File: docs/performance/tune-performance.md (hunk 6 above)
Comment (RandySheriffH) [reply docs/performance/tune-performance.md]: Added separate section for NUMA.

**Comment by RandySheriffH** [comment]
> @RandySheriffH please stage your changes for visual preview - https://github.com/microsoft/onnxruntime/wiki/Update-ONNX-Runtime-Docs#onnxruntimeaidocs

https://randysheriffh.github.io/onnxruntime/

File: docs/execution-providers/Azure-ExecutionProvider.md [hunk 8]
```diff
@@ -0,0 +1,74 @@
+---
+title: Azure
+description: Instructions to infer an ONNX model remotely with an Azure endpoint
+parent: Execution Providers
+nav_order: 12
+redirect_from: /docs/reference/execution-providers/Azure-ExecutionProvider
+---
+
+# Azure Execution Provider (Preview)
```
Comment (natke) [inline docs/execution-providers/Azure-ExecutionProvider.md:9]: Do we want to call this document Azure Execution Provider? I'm thinking about what people will search for to find it. Would Cloud execution provider make more sense?

File: docs/execution-providers/Azure-ExecutionProvider.md [hunk 9]
```diff
@@ -0,0 +1,74 @@
...
+The Azure Execution Provider enables ONNX Runtime to visit an remote endpoint for inferenece, the endpoint must be deployed beforehand.
+Azure Execution Provider is in preview stage, all API(s) and usage are subjuct to change.
+
+One use case for Azure Execution Provider is small-big models. E.g. A smaller model deployed on edge device for faster inference,
+while a bigger model deployed on Azure for higher precision, with Azure Execution Provider, a switch between the two could be easily achieved.
+Note that the two models are expected to have same inputs and outputs.
+
+## Limitations
+
+To consume the endpoint, ONNX Runtime must load the model locally, it could be the same model deployed to Azure, or another model that has exactly same inputs and outputs.
```
Comment (natke) [inline docs/execution-providers/Azure-ExecutionProvider.md:20]: The intro para here does not feel like a limitation (and makes the EP look more limited than it is). Can we move the para outside of the limitations section?

File: docs/execution-providers/Azure-ExecutionProvider.md (hunk 8 above)
Comment (pranavsharma) [reply docs/execution-providers/Azure-ExecutionProvider.md:9]: @jeffmend specifically wanted to call this Azure EP. We'd called it Cloud EP before :-).

Comment (jeffmend) [reply docs/execution-providers/Azure-ExecutionProvider.md:9]: Yes, this was an intentional choice to go with Azure EP in the docs.  would be good if it came up in searches.

Comment (natke) [reply docs/execution-providers/Azure-ExecutionProvider.md:9]: Ok, perhaps we should also (not in this PR, but later) add a tutorial that uses more scenario specific terminology and that could link to this doc

File: docs/execution-providers/Azure-ExecutionProvider.md (hunk 9 above)
Comment (RandySheriffH) [reply docs/execution-providers/Azure-ExecutionProvider.md]: Moved to description.

**Review by natke — APPROVED** [review]

File: index.html [hunk 10]
```diff
@@ -249,6 +249,8 @@ <h3 id="selectHardwareAcceleration">Hardware Acceleration</h3>
 															<span>Vitis AI (Preview)</span></div>
 														<div class="col-lg-3 col-md-3 r-option version" role="option" tabindex="-1" aria-selected="false" id="XNNPACK">
 															<span>XNNPACK (Preview)</span></div>
+														<div class="col-lg-3 col-md-3 r-option version" role="option" tabindex="-1" aria-selected="false" id="Azure">
+															<span>Azure (Preview)</span></div>
```
Comment (faxu) [inline index.html:253]: This should be alphabetical in the "preview" portion of the list. So it should be after "ArmNN"

File: docs/execution-providers/Azure-ExecutionProvider.md [hunk 11]
```diff
@@ -0,0 +1,74 @@
+---
+title: Azure
```
Comment (faxu) [inline docs/execution-providers/Azure-ExecutionProvider.md:2]: Suggest to name this "Cloud - Azure" for consistency.

File: docs/execution-providers/Azure-ExecutionProvider.md [hunk 12]
```diff
@@ -0,0 +1,74 @@
...
+description: Instructions to infer an ONNX model remotely with an Azure endpoint
+parent: Execution Providers
+nav_order: 12
```
Comment (faxu) [inline docs/execution-providers/Azure-ExecutionProvider.md:5]: Should be 11, and community-maintained and add-execution-provider pages need to be shifted to 12 and 13.

File: docs/execution-providers/Azure-ExecutionProvider.md [hunk 13]
```diff
@@ -0,0 +1,74 @@
...
+redirect_from: /docs/reference/execution-providers/Azure-ExecutionProvider
```
Comment (faxu) [inline docs/execution-providers/Azure-ExecutionProvider.md:6]: I think this line is not needed since this file never existed at previous location

Comment (RandySheriffH) [reply docs/execution-providers/Azure-ExecutionProvider.md]: Removed.

File: docs/execution-providers/Azure-ExecutionProvider.md (hunk 12 above)
Comment (RandySheriffH) [reply docs/execution-providers/Azure-ExecutionProvider.md]: Done.

File: docs/execution-providers/Azure-ExecutionProvider.md (hunk 11 above)
Comment (RandySheriffH) [reply docs/execution-providers/Azure-ExecutionProvider.md]: Done.

File: index.html (hunk 10 above)
Comment (RandySheriffH) [reply index.html:255]: Moved.

**Pull request merged**